waspc/data/Generator/templates/server/src/auth/providers/local/signup.ts

```diff
@@ -1,26 +1,39 @@
 {{={= =}=}}
 import { handleRejection } from '../../../utils.js'
-import { createUser } from '../../utils.js'
-import { ensureValidUsername, ensurePasswordIsPresent, ensureValidPassword } from '../../validation.js'
+import {
+  createAuthWithUser,
+  sanitizeAndSerializeProviderData,
+} from '../../utils.js'
+import {
+  ensureValidUsername,
+  ensurePasswordIsPresent,
+  ensureValidPassword,
+} from '../../validation.js'
 import { validateAndGetAdditionalFields } from '../../utils.js'
 
 export default handleRejection(async (req, res) => {
-  const userFields = req.body || {}
-  ensureValidArgs(userFields)
+  const fields = req.body ?? {}
+  ensureValidArgs(fields)
 
-  const additionalFields = await validateAndGetAdditionalFields(userFields)
+  const userFields = await validateAndGetAdditionalFields(fields)
 
-  await createUser({
-    ...additionalFields,
-    username: userFields.username,
-    password: userFields.password,
+  const providerData = await sanitizeAndSerializeProviderData<'username'>({
+    password: fields.password,
   })
+  await createAuthWithUser(
+    'username',
+    fields.username,
+    providerData,
+    // Using any here because we want to avoid TypeScript errors and
+    // rely on Prisma to validate the data.
```

Comment:
Prisma will validate it because these will get save into User entity, right?

Reply:
Prisma will throw an error during saving it in the DB if something is wrong.